sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DocumentQueryExecutionContextFactory.java

```diff
@@ -358,6 +361,37 @@ public static <T> Flux<? extends IDocumentQueryExecutionContext<T>> createSpecia
 
         boolean getLazyFeedResponse = queryInfo.hasTop();
 
+        // We need to compute the optimal initial age size for non-streaming order-by queries
+        if (queryInfo.hasNonStreamingOrderBy() && Configs.getMaxItemSizeForVectorSearchEnabled()) {
+            // Validate the TOP or LIMIT for non-streaming order-by queries
+            if (!queryInfo.hasTop() && !queryInfo.hasLimit() && queryInfo.getTop() < 0 && queryInfo.getLimit() < 0) {
```

Comment:
I think if Limit is used you effectively need to retrieve Offset + Limit records - so the MaxBufferCount needs to be <= Offset + Limit - not just limit?

Reply:
Added the code logic for this.